File: testapp/src/main/java/com/juanimoli/strictmodenotifier/testapp/LeakedClosableObjectsActivity.java
package com.juanimoli.strictmodenotifier.testapp;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import java.io.BufferedWriter;
import java.io.OutputStreamWriter;

public class LeakedClosableObjectsActivity extends AppCompatActivity {

    public static void start(Context context) {
        Intent starter = new Intent(context, LeakedClosableObjectsActivity.class);
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        try {
            OutputStreamWriter output_writer =
                    new OutputStreamWriter(openFileOutput("FILETEST", MODE_PRIVATE));
            BufferedWriter writer = new BufferedWriter(output_writer);
            writer.write("Hi This is my File to test StrictMode");
            //writer.close();
        } catch (Exception e) {
            e.printStackTrace();
        }

        finish();
    }

    private void fireLeakedClosableObjects() {

    }
}

File: testapp/src/main/java/io/github/juanimoli/strictmode/notifier/ui/LeakedClosableObjectsActivity.kt
package io.github.juanimoli.strictmode.notifier.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import java.io.BufferedWriter
import java.io.OutputStreamWriter

class LeakedClosableObjectsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        try {
            val outputWriter = OutputStreamWriter(openFileOutput("FILETEST", MODE_PRIVATE))
            val writer = BufferedWriter(outputWriter)
            writer.write("Hi This is my File to test StrictMode")
            //writer.close();
        } catch (e: Exception) {
            e.printStackTrace()
        }

        finish()
    }

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, LeakedClosableObjectsActivity::class.java)
            context.startActivity(starter)
        }
    }
}
